builtin/credential/approle/backend.go

```diff
@@ -73,6 +74,10 @@ func Backend(conf *logical.BackendConfig) (*backend, error) {
 	// a superfluous number of locks directly proportional to the number of RoleID
 	// and SecretIDs. These locks can be accessed by indexing based on the first two
 	// characters of a randomly generated UUID.
+	if err = locksutil.CreateLocks(b.roleLocksMap, 256); err != nil {
```

Comment:
> Since map is a reference type, locksutil might have to return the created map reference back to the caller, I guess. This may not intuitive for callers of this util.

Oh, good point.

> Creating map of fixed size in the utility might reduce flexibility to add more items to the map outside of this utility function (like how we are adding "custom" lock).

Keep in mind map size is not a limit, it's just an initial capacity. But I agree, keep it out of locksutil.